authentication/src/main/java/uk/gov/android/authentication/ILoginSession.kt

```diff
@@ -0,0 +1,14 @@
+package uk.gov.android.authentication
+
+import android.content.Context
+import android.content.Intent
+import net.openid.appauth.TokenResponse
+
+interface ILoginSession {
+    fun present(
+        configuration: LoginSessionConfiguration
+    )
+
+    fun init(context: Context): ILoginSession
+    fun finalise(intent: Intent, callback: (tokens: TokenResponse) -> Unit)
```

Comment:
I dunno how I feel about exposing these third party dependencies as function parameters.

If that's the application or activity context in `init()`, we can make that a Hilt injectable thing within an implementation's constructor so passing the context isn't required.

Reply:
This has been changed and refactored as you have suggested in the actual seperate repo here: https://github.com/govuk-one-login/mobile-android-authentication/blob/main/app/src/main/java/uk/gov/android/authentication/LoginSession.kt

This code will be removed from this repo soon